src/hotspot/share/gc/shenandoah/shenandoahAgeCensus.hpp

```diff
@@ -165,11 +167,18 @@ class ShenandoahAgeCensus: public CHeapObj<mtGC> {
   };
 
   ShenandoahAgeCensus();
+  ShenandoahAgeCensus(uint max_workers);
+  ~ShenandoahAgeCensus();
 
   // Return the local age table (population vector) for worker_id.
   // Only used in the case of (ShenandoahGenerationalAdaptiveTenuring && !ShenandoahGenerationalCensusAtEvac)
   AgeTable* get_local_age_table(uint worker_id) {
-    return (AgeTable*) _local_age_table[worker_id];
+    return _local_age_table[worker_id];
+  }
+
+  // Return true if this age is above the tenuring threshold.
+  bool is_tenurable(uint age) const {
+    return age > tenuring_threshold();
```

Comment:
Another key change. This prevents the algorithm from seeing higher mortality rates caused by promotions in the cohort at tenuring age.